gtsfm/utils/geometry_comparisons.py

```diff
@@ -46,30 +46,21 @@ def align_poses(input_list: List[Pose3], ref_list: List[Pose3]) -> List[Pose3]:
         transformed poses which have the same origin and scale as reference
             (now living in world-2 frame)
     """
-    w1Ti0 = input_list[0]
-    i0Tw1 = w1Ti0.inverse()
-    w2Ti0_ = ref_list[0]
-    # origin transform -- map the origin of the input list to the reference list
-    w2Tw1 = w2Ti0_.compose(i0Tw1)
-    
-    # origin shifted list
-    input_shifted_list = [w2Tw1.compose(w1Ti) for w1Ti in input_list]
-
-    # get distances w.r.t origin for both the list to compute the scale
-    w2Ti0 = input_shifted_list[0] # set this as origin
+    # match the scales first
```

Comment:
what's the change we're making here?

Reply:
I was first matching the first pose, and then aligning the scale (by using the median value). This creates and error. 

Example: we have 5 poses on the circle. The circle center is at the origin. Now a scaled version of this scene is the one where we have doubled the radius of the circle.

Earlier, the align_pose function first matched the 0th pose (i.e. set the same rotation and translation) and then scaled the scene. That logic will not work. So now I first match the scale and then align the 0th pose.

I had a unit test for this, but it did not run earlier because I messed up the unit test name to create two tests with the same name.

Reply:
I feel like either of them should work since the scale is with respect to the first pose and not the world frame.

Reply:
Nope. I use the median of scale of all (i, 0) pairs, i.e. translation magnitude ratio for all poses w.r.t. 0th pose.

I'll replace it with whatever we have in Zhang18iros_evaluation. But for now I'll leave this as is.

Reply:
This is a nice fix @ayushbaid . I'll add another test case that now passes with the fixed code. Can we put the geometry comparison fix in its own PR though to make the PRs simpler? Then we can accept it immediately while we discuss this other stuff

Reply:
Yeah the test case was already there. I mistakenly named it w/ name of another unit test and hence it was running earlier. It will run now.